/* Copyright 2023 The FedLearner Authors. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package fedlearner_webconsole.proto;

enum TicketType {
    UNKOWN_TYPE = 0;
    CREATE_NODE = 1;
    CREATE_PARTICIPANT = 2;
    CREATE_PROJECT = 3;
    PUBLISH_DATASET = 4;
    CREATE_PROCESSED_DATASET = 5;
    CREATE_MODELJOB_GROUP = 6;
    TK_CREATE_TRUSTED_JOB_GROUP = 7;
    TK_CREATE_TRUSTED_EXPORT_JOB = 8;
}

enum ReviewStrategy {
    MANUAL = 0;
    AUTO = 1;
}

enum ReviewStatus {
    PENDING = 0;
    APPROVED = 1;
    DECLINED = 2;
}

message TicketDetails {
    // uuid for review ticket resource.
    string uuid = 1;
}

message Ticket {
    int64 id = 1;
    string uuid = 2;
    // Username of creator
    string creator_username = 3;
    string coordinator_pure_domain_name = 4;
    TicketType type = 5;
    string reviewer_username = 6;
    ReviewStrategy review_strategy = 7;
    ReviewStatus status = 8;
    TicketDetails details = 9;
    // Timestamp in seconds
    int64 created_at = 10;
    // Timestamp in seconds
    int64 updated_at = 11;
    // Timestamp in seconds
    int64 deleted_at = 12;
}
